### PR TITLE
Issue151 integration

### PR DESCRIPTION
**Pull request by pminak**

### CODE REVIEW

**Comment by timspainNERSC** [comment]
It looks like `ubuntu-latest` is not the latest version of Ubuntu, which does have Eigen 3.4.

Indeed, it is still 20.04LTS `focal`, which only has Eigen version 3.3.7.

File: dynamics/CMakeLists.txt [hunk 1]
```diff
@@ -1,113 +1,16 @@
...
-
-################################################## Compile the test cases
+add_subdirectory(applications)
 add_subdirectory(tests)
 
-################################################## Compile larger application examples
-add_subdirectory(applications)
 
+set(NextsimSources
+    "${NextsimSources}"
```
Comment (timspainNERSC) [inline dynamics/CMakeLists.txt:11]: If you are not (yet) changing the entries in `NextsimSources`, these lines should be removed as redundant. (Also lines 14-16)

File: dynamics/src/CMakeLists.txt [hunk 2]
```diff
@@ -7,12 +12,16 @@ set (INC "${CMAKE_CURRENT_SOURCE_DIR}/include")
...
-INCLUDE_DIRECTORIES(${Nextsim_INCLUDE_DIRS})
-
-add_library(${NextsimDGLib} ${Nextsim_Sources})
-
-target_link_libraries (${NextsimDGLib} Eigen3::Eigen)
-
 
+set(NextsimSources
+    "${NextsimSources}"
+    "${Sources}"
```
Comment (timspainNERSC) [inline dynamics/src/CMakeLists.txt:18]: Again, remove these set( PARENT_SCOPE) commands.

**Review by timspainNERSC — CHANGES_REQUESTED** [review]
I don't think we should include the set(PARENT_SCOPE) commands until they are actually being used.

**Comment by einola** [comment]
Looks like @pminak has addressed the issues raised by @timspainNERSC - right guys?

File: dynamics/src/CMakeLists.txt (hunk 2 above)
Comment (timspainNERSC) [reply dynamics/src/CMakeLists.txt:18]: Okay, fine.

**Review by timspainNERSC — APPROVED** [review]

**Pull request merged**